packages/plugins/typescript/urql-graphcache/src/index.ts

```diff
@@ -0,0 +1,244 @@
+    const { fields } = subscriptionType.astNode;
+    fields.forEach(fieldNode => {
+      const argsName = `Mutation${capitalize(fieldNode.name.value)}Args`;
+      const type = unwrapType(fieldNode.type);
+      updaters.push(
+        `${fieldNode.name.value}?: GraphCacheUpdateResolver<{ ${fieldNode.name.value}: ${constructType(
+          type,
+          schema
+        )} }, ${argsName}>`
+      );
+    });
+  }
+
+  return updaters;
+}
+
+function getMutationUpdaterConfig(schema: GraphQLSchema, mutationName: string) {
+  const updaters = [];
+  const typemap = schema.getTypeMap();
+  const mutationType = typemap[mutationName];
+
+  if (mutationType.astNode.kind === Kind.OBJECT_TYPE_DEFINITION) {
+    const { fields } = mutationType.astNode;
+    fields.forEach(fieldNode => {
+      const argsName = `Mutation${capitalize(fieldNode.name.value)}Args`;
+      const type = unwrapType(fieldNode.type);
+      updaters.push(
+        `${fieldNode.name.value}?: GraphCacheUpdateResolver<{ ${fieldNode.name.value}: ${constructType(
+          type,
+          schema
+        )} }, ${argsName}>`
+      );
+    });
+  }
+
+  return updaters;
+}
+
+function getOptimisticUpdatersConfig(schema: GraphQLSchema, mutationName: string) {
+  const optimistic = [];
+  const typemap = schema.getTypeMap();
+  const mutationType = typemap[mutationName];
+
+  if (mutationType.astNode.kind === Kind.OBJECT_TYPE_DEFINITION) {
+    const { fields } = mutationType.astNode;
+    fields.forEach(fieldNode => {
+      const argsName = `Mutation${capitalize(fieldNode.name.value)}Args`;
+      const type = unwrapType(fieldNode.type);
+      const outputType = constructType(type, schema);
+      optimistic.push(`${fieldNode.name.value}?: GraphCacheOptimisticMutationResolver<${argsName}, ${outputType}>`);
+    });
+  }
+
+  return optimistic;
+}
+
+function createCacheGeneric() {
+  return `export type GraphCacheConfig = {
+  updates: GraphCacheUpdaters;
+  keys: GraphCacheKeysConfig;
+  optimistic: GraphCacheOptimisticUpdaters;
+  resolvers: GraphCacheResolvers;
+}`;
+}
+
+export const plugin: PluginFunction<UrqlGraphCacheConfig, Types.ComplexPluginOutput> = (schema: GraphQLSchema) => {
+  const mutationName = schema.getMutationType()?.name;
+  const subscriptionsName = schema.getSubscriptionType()?.name;
+
+  const keys = getKeysConfig(schema);
+  const resolvers = getResolversConfig(schema);
+  let mutationUpdaters, subscriptionUpaters, optimisticUpdaters;
+  if (mutationName) {
+    mutationUpdaters = getMutationUpdaterConfig(schema, mutationName);
+    optimisticUpdaters = getOptimisticUpdatersConfig(schema, mutationName);
+  }
+
+  if (subscriptionsName) {
+    subscriptionUpaters = getSubscriptionUpdatersConfig(schema, subscriptionsName);
+  }
+
+  return {
+    prepend: [imports],
+    content: [
+      `type RequireFields<T, K extends keyof T> = { [X in Exclude<keyof T, K>]?: T[X] } & { [P in K]-?: NonNullable<T[P]> };`,
```

Comment:
Can we check somehow whether another plugin already adds this?

Reply:
At the moment, we can't, we are currently trying to figure out how to allow interaction between plugins, so it will be simpler to check what other plugins are doing. 
If this type could conflict (I think we use it in `typescript-resolvers` only, no? does it makes sense to use both together?), maybe it can be named differently?

Reply:
I think given this we may just want to name it differently? I think it's fine to make a more specialised type for this generator that explicitly only works with `'__typename'`?